### DISCUSSION TITLE
Is it possible that get data in A loader from B loader

### DISCUSSION THREAD

**Discussion by WreewanMorhee** [remix-run/remix#7751] [Q&A] [answered]
like title 
is it possible ??

**Comment by alexanderson1993** [comment, marked as answer]
If you want a parent loader's data, you might consider looking at something like useMatches to get the data of the parent route. https://remix.run/docs/en/main/hooks/use-matches

Otherwise, DataLoader should meet your needs. It's a request cache that uses a single request to get data, so if your loaders are fetching from the same data source, it deduplicates the fetch. Here's an example showing how to use it with Remix. https://github.com/remix-run/examples/tree/main/dataloader

**Reply by sergiodxa** [reply]
You can also use React Query's core (TanStack Query) as a DataLoader https://sergiodxa.com/tutorials/use-tanstack-query-to-share-data-between-remix-loaders.

And for useMatches, remember it's only available in React components, so it can work to share data in the UI but not in the loader itself.

**Reply by WreewanMorhee** [reply]
got it 
thank u all
